### ISSUE TITLE
Upgrade RUST used in bpf-tools

### ISSUE THREAD

**Issue by great9** [solana-labs/solana#24321]
#### Problem
One of the crates that I'm using in my contract requires rust newer than 1.56.0-dev.
I've already submitted an issue to build-bpf which was closed a couple of days later.

```
error: package `uint v0.9.3` cannot be built because it requires rustc 1.56.1 or newer, while the currently active rustc version is 1.56.0-dev
```

I'm using the latest solana release right now (1.10.8) through
```
sh -c "$(curl -sSfL https://release.solana.com/v1.10.8/install)"
```

which includes rustc binary version 1.56.0-dev
```
~/.local/share/solana/install/releases/1.10.8/solana-release/bin/sdk/bpf/dependencies/bpf-tools/rust/bin/rustc --version
rustc 1.56.0-dev
```

my rustc version:
```
$ rustc --version
rustc 1.60.0 (7737e0b5c 2022-04-04)
```

#### Proposed Solution
Option 1: upgrade rustc in bpf-tools
Option 2: symlink "... solana-release/bin/sdk/bpf/dependencies/bpf-tools/rust/bin/rustc" to the rustup-installed version. Haven't tried this.

**Comment by dmakarov** [comment]
bpf-tools already upgraded to rust 1.59. Master branch of solana already uses the upgraded bpf-tools. The 1.10 branch of solana is still pending the backport of the upgrade.

**Comment by great9** [comment]
> bpf-tools already upgraded to rust 1.59. Master branch of solana already uses the upgraded bpf-tools. The 1.10 branch of solana is still pending the backport of the upgrade.

Proposed solution?

**Comment by dmakarov** [comment]
Actually upgraded tools were merged to 1.10 branch #24194. Rust 1.59 will be available in the next minor release of Solana.

**Comment by monomadic** [comment]
I hate to be 'that guy' but perhaps a priority should be focused around being more canonical to rust. Cargo can install a bpf toolchain, with the current version of rust (1.60). I realise testing needs to take place, but as exemplified here, things fall apart quickly with a custom install path - and I'm not a fan of curling a shell script.
